core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -268,10 +270,12 @@ public synchronized void gainPrimacy() throws IOException {
       resume();
     }
 
+    // If the tailing thread has crashed, the standby master will crash here
+    // instead of gaining primacy
     mTailerThread.awaitTermination(true);
```

Comment:
yes because you throw an exception 
I'm thinking if it will become cleaner if you let awaitTermination() throw a checked exception and let the callers handler them explicitly?

Reply:
Yea Arthur is proposing the same, I guess we'll just converge to that. Logged in https://github.com/Alluxio/alluxio/issues/16381 and I figure we should address that systematically, in a bit(not in this PR).

Reply:
Checked exceptions should be used if the exception can be recovered from, and the caller knows what to do with it. An exception thrown on this line reflects a catastrophic journal failover (catastrophic as in it cannot be recovered from). In this instance, an unchecked exception propagated to the very top may make more sense: callers would not know what to do with a checked exception anyway.